### DISCUSSION TITLE
可以推到pypi吗？

### DISCUSSION THREAD

**Discussion by zqabc** [Django-Simple-API/django-simple-api#42] [General]
如题

**Comment by abersheeran** [comment]
目前来看代码是无 BUG 了，等到文档完成之后就会推送代码到 PyPi。如果你十分急切的话，可以提 PR 帮助我们完成文档。

**Reply by zqabc** [reply]
哪些文档需要完成？我可以帮忙
另外，describe_response我觉得少了自动验证输出的功能，目前只有生成文档。
我想了下是否可以这样修改?
1. set describe_response to request
```
def _verify_params(
    handler: HTTPHandler, request: HttpRequest, may_path_params: Dict[str, Any]
) -> Dict[str, Any]:
    # set describe_response to request
    request.__describe_response__ = getattr(handler, "__responses__", None)
```

2. django middleware
```
def process_response(self, request, response):
    if isinstance(response, tuple):
        status, result = response
        describe_response = getattr(request, '__describe_response__', None)
        if describe_response:
            for k, v in describe_response.items():
                if status == k:
                    v['content'].parse_obj(result)
                    break
```
3. view
return 200, {"msg":"good"}

**Reply by abersheeran** [reply]
确实在设计如何利用起来这个模型，如果仅生成文档就太浪费了。欢迎相关的 PR，但你目前给出的 demo 可适应能力太差，响应不止是 JSON 的，也可能有图片、视频、文本等能被 OpenAPI 描述的。

另外，你这是同时做了两件事，转换 view 返回值到 HttpResponse、校验返回结果是否符合设计，应当拆分成两个中间件来实现，这样使用的时候就可以自由组合多个中间件来选择需要的功能。

**Reply by zqabc** [reply]
sorry，最近太忙了，我的场景全是响应JSON的，所以只按JSON处理了，其他的部分有时间我再想下。

**Reply by abersheeran** [reply]
没关系，有时间、有兴趣的时候再来做吧

**Reply by abersheeran** [reply]
随时欢迎。

**Comment by waitan2018** [comment]
目前只有`describe_extra_docs` 和 `Support for JSON requests` 的文档没有写完，当然您也可以补充其他看起来不完美的地方。

**Reply by waitan2018** [reply]
我刚刚发现应该还缺少上传文件和图片(`UploadFile`/`UploadImage`)的示例。

**Reply by waitan2018** [reply]
你可以在 docs 里修改 md 文档，然后在根目录使用命令 `mkdocs serve` 预览效果。

**Comment by waitan2018** [comment]
emm，时隔n久，突然想起来，这个库已经发布到 pypi 了，这里说下
